pkg/devices/gen_net.go

```diff
@@ -82,8 +82,13 @@ func NewGenNetDevice(deviceID string, dt types.DeviceType, isRdma bool) (*GenNet
 	}
 
 	linkType := ""
-	if len(ifName) > 0 {
-		la, err := utils.GetNetlinkProvider().GetLinkAttrs(ifName)
+	linkTypeProviderDevice := ifName
+	// If interface name is not available, derive link type from PF
+	if linkTypeProviderDevice == "" {
+		linkTypeProviderDevice = pfName
+	}
+	if linkTypeProviderDevice != "" {
```

Comment:
I don't think we need this if state here.
we have two cases only
1. vf exist use vf
2. vf doesn't exist is pf 

there is no other case so it can't be empty

Reply:
There is, in case of VF passthrough, and device plugin runs in VM.

we have no PF and VF Ifname may not exist if its already assigned to a container

Reply:
right!